x-pack/lib/filters/geoip/database_manager.rb

```diff
@@ -302,6 +302,11 @@ def metric=(metric)
     @metric = metric
   end
 
+  def metric
+    # Fallback when testing plugin and no metric collector are correctly configured.
+    @metric.nil? ? LogStash::Instrument::NamespacedNullMetric.new: @metric
```

Comment:
if this is just for tests, could we mock or inject the NullMetric?
it looks weird that we instantiate a new one (which is fine for tests), 
should be also fine setting up a `@metric ||= LogStash::Instrument::NamespacedNullMetric.new` during initialize

Reply:
It is possible to mock in plugin test but would need to mock and prepare in a way that may look unnatural. So doing in `#metric` make things simpler.  I took the suggestion `@metric ||= `, thank you.